src/huggingface_hub/hf_file_system.py

```diff
@@ -649,6 +654,58 @@ def _upload_chunk(self, final: bool = False) -> None:
             )
 
 
+class HfFileSystemStreamFile(fsspec.spec.AbstractBufferedFile):
+    def __init__(
+        self, fs: HfFileSystem, path: str, revision: Optional[str] = None, block_size=0, cache_type="none", **kwargs
+    ):
+        if block_size != 0:
+            raise ValueError(f"HfFileSystemStreamFile only supports block_size=0 but got {block_size}")
+        if cache_type != "none":
+            raise ValueError(f"HfFileSystemStreamFile only supports cache_type='none' but got {cache_type}")
+        try:
+            self.resolved_path = fs.resolve_path(path, revision=revision)
+        except FileNotFoundError as e:
+            if "w" in kwargs.get("mode", ""):
+                raise FileNotFoundError(
+                    f"{e}.\nMake sure the repository and revision exist before writing data."
+                ) from e
+        self.details = {"name": self.resolved_path.unresolve(), "size": None}
+        super().__init__(fs, self.resolved_path.unresolve(), block_size=block_size, cache_type=cache_type, **kwargs)
+        self.r: Optional[Response] = None
+        self.fs: HfFileSystem
+
+    def seek(self, loc, whence=0):
+        if loc == 0 and whence == 1:
+            return
+        if loc == self.loc and whence == 0:
+            return
+        raise ValueError("Cannot seek streaming HF file")
+
+    def read(self, length=-1):
+        if self.r is None:
+            url = hf_hub_url(
+                repo_id=self.resolved_path.repo_id,
+                revision=self.resolved_path.revision,
+                filename=self.resolved_path.path_in_repo,
+                repo_type=self.resolved_path.repo_type,
+                endpoint=self.fs.endpoint,
+            )
+            self.r = http_backoff("GET", url, headers=self.fs._api._build_hf_headers(), stream=True)
```

Comment:
We need to ensure this response is closed (after reading the content or on an error), as advised in the official `requests` docs: https://requests.readthedocs.io/en/latest/user/advanced/#body-content-workflow

Additionally, we should account for the 5XX HTTP errors (as done in https://github.com/huggingface/huggingface_hub/pull/1981)

Reply:
Good catch !

The connection already closed after reading the content, I'll add a try/except when there is an error to close it.